HLTrigger/Muon/src/HLTMuonDimuonL3Filter.cc

```diff
@@ -161,19 +165,63 @@ HLTMuonDimuonL3Filter::hltFilter(edm::Event& iEvent, const edm::EventSetup& iSet
    Handle<RecoChargedCandidateCollection> mucands;
    if (saveTags()) filterproduct.addCollectionTag(candTag_);
    iEvent.getByToken(candToken_,mucands);
+
+   // Test to see if we can use L3MuonTrajectorySeeds:
+   if (mucands->size()<1) return false;
+   auto tk = (*mucands)[0].track();
+   bool useL3MTS=false;
+
+   if (tk->seedRef().isNonnull()){
+	   auto a = dynamic_cast<const L3MuonTrajectorySeed*>(tk->seedRef().get());
```

Comment:
The dynamic_cast doesn't work if we are not using L3MuonTrajectorySeed ... if we dont use a L3MuonTrajectorySeed then we are using the normal TrajectorySeeds.